notify/src/windows.rs

```diff
@@ -19,23 +19,34 @@ use std::ptr;
 use std::slice;
 use std::sync::{Arc, Mutex};
 use std::thread;
-use windows_sys::Win32::Foundation::{
-    CloseHandle, ERROR_OPERATION_ABORTED, HANDLE, INVALID_HANDLE_VALUE, WAIT_OBJECT_0,
-};
-use windows_sys::Win32::Storage::FileSystem::{
-    CreateFileW, ReadDirectoryChangesW, FILE_ACTION_ADDED, FILE_ACTION_MODIFIED,
-    FILE_ACTION_REMOVED, FILE_ACTION_RENAMED_NEW_NAME, FILE_ACTION_RENAMED_OLD_NAME,
-    FILE_FLAG_BACKUP_SEMANTICS, FILE_FLAG_OVERLAPPED, FILE_LIST_DIRECTORY,
-    FILE_NOTIFY_CHANGE_ATTRIBUTES, FILE_NOTIFY_CHANGE_CREATION, FILE_NOTIFY_CHANGE_DIR_NAME,
-    FILE_NOTIFY_CHANGE_FILE_NAME, FILE_NOTIFY_CHANGE_LAST_WRITE, FILE_NOTIFY_CHANGE_SECURITY,
-    FILE_NOTIFY_CHANGE_SIZE, FILE_NOTIFY_INFORMATION, FILE_SHARE_DELETE, FILE_SHARE_READ,
-    FILE_SHARE_WRITE, OPEN_EXISTING,
-};
-use windows_sys::Win32::System::Threading::{
-    CreateSemaphoreW, ReleaseSemaphore, WaitForSingleObjectEx,
-};
-use windows_sys::Win32::System::WindowsProgramming::INFINITE;
-use windows_sys::Win32::System::IO::{CancelIo, OVERLAPPED};
+
+#[cfg(all(feature = "windows-sys", not(feature = "slim_windows")))]
+mod sys {
+    pub use windows_sys::Win32::Foundation::{
+        CloseHandle, ERROR_OPERATION_ABORTED, HANDLE, INVALID_HANDLE_VALUE, WAIT_OBJECT_0,
+    };
+    pub use windows_sys::Win32::Storage::FileSystem::{
+        CreateFileW, ReadDirectoryChangesW, FILE_ACTION_ADDED, FILE_ACTION_MODIFIED,
+        FILE_ACTION_REMOVED, FILE_ACTION_RENAMED_NEW_NAME, FILE_ACTION_RENAMED_OLD_NAME,
+        FILE_FLAG_BACKUP_SEMANTICS, FILE_FLAG_OVERLAPPED, FILE_LIST_DIRECTORY,
+        FILE_NOTIFY_CHANGE_ATTRIBUTES, FILE_NOTIFY_CHANGE_CREATION, FILE_NOTIFY_CHANGE_DIR_NAME,
+        FILE_NOTIFY_CHANGE_FILE_NAME, FILE_NOTIFY_CHANGE_LAST_WRITE, FILE_NOTIFY_CHANGE_SECURITY,
+        FILE_NOTIFY_CHANGE_SIZE, FILE_NOTIFY_INFORMATION, FILE_SHARE_DELETE, FILE_SHARE_READ,
+        FILE_SHARE_WRITE, OPEN_EXISTING,
+    };
+    pub use windows_sys::Win32::System::Threading::{
+        CreateSemaphoreW, ReleaseSemaphore, WaitForSingleObjectEx,
+    };
+    pub use windows_sys::Win32::System::WindowsProgramming::INFINITE;
+    pub use windows_sys::Win32::System::IO::{CancelIo, OVERLAPPED};
+}
+#[cfg(all(feature = "windows-sys", not(feature = "slim_windows")))]
+use sys::*;
+
+#[cfg(any(not(feature = "windows-sys"), feature = "slim_windows"))]
```

Comment:
This is sadly a slippery slope. People use no-default-features to specify kqueue as macos backend. By this definition they automatically get the slim backend, which is unintentional (and got me wondering why the CI passed in the first place).

Reply:
I've already had a working change to make it a requirement, but then we'd have to bump the major version as everyone has to migrate their features. So that's not really a solution.

Reply:
I originally had it on just whether they had the slim feature enabled, but if they disable default features and don't enable slim it would fail to compile, but if you're OK with that I can do that instead.